### ISSUE TITLE
Allow side-effect imports with `no-empty-pattern` enabled

### ISSUE THREAD

**Issue by MadLittleMods** [eslint/eslint#4273]
Allow side-effect imports with  [`no-empty-pattern`](https://github.com/eslint/eslint/blob/master/docs/rules/no-empty-pattern.md)

```js
/* eslint no-empty-pattern: 2 */
import 'babel/polyfill';
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
I cant reproduce this with online demo: http://eslint.org/demo/

**Comment by MadLittleMods** [comment]
@gyandeeps Thanks, I had path problems and was pointing to and old `v0.24.1` copy from a different node install.

Everything seems to work appropriately :)
